firestore-send-email/functions/src/index.ts

```diff
@@ -95,15 +96,51 @@ async function processCreate(snap: FirebaseFirestore.DocumentSnapshot) {
   });
 }
 
-async function preparePayload(payload: QueuePayload): Promise<QueuePayload> {
+export const findUser = async function(
+  recipients: string | string[]
+): Promise<UserRecord> | undefined {
+  const isString: boolean = typeof recipients === "string";
+
+  if (isString && !recipients.length) return null;
+  if (isString && (recipients as string).split(",").length > 1) return null;
+  if (!isString && recipients.length > 1) return null;
+
+  const recipient: string = isString ? (recipients as string) : recipients[0];
```

Comment:
This was limited so that personalised emails could not be sent to multiple recipients within the same email.

For example, the following configuration would personalise to multiple users named in the `to` list:

```
{to: ["user1@email.com", "user2@email.com", "user3@email.com"]}
```

Personalisation would be based on single  recipients based on each `document` in the `mails` collection for example.